mikupad.html

```diff
@@ -7171,6 +7172,35 @@ <h1>Oops! Something went wrong.</h1>
 	}
 
 	async function predict(prompt = finalPromptText, chunkCount = promptChunks.length, callback = undefined, abortController = undefined, invalidatesUndo = false, customParams = {}) {
+
+		// If a callback is provided, this is an internal call, so we skip the in-filling logic.
+		if (!callback) {
+			const elem = promptArea.current;
+			const cursorPosition = elem.selectionStart;
+			const isSelection = elem.selectionEnd > cursorPosition;
+			const isAtEnd = cursorPosition === elem.value.length && !isSelection;
+			const hasExistingPlaceholder = elem.value.includes('{predict}') || elem.value.includes('{fill}');
+
+			// If the cursor is not at the end and there's no placeholder,
+			// we treat it as a fill-in-the-middle request from the cursor.
+			if (generateFromCursor && !isAtEnd && !hasExistingPlaceholder) {

```

Comment:
After the above change, checking here won't be necessary:
```suggestion
			if (!isAtEnd && !hasExistingPlaceholder) {
```